### PR TITLE
xds: ignore endpoints with health_status != HEALTHY or UNKNOWN

### PR DESCRIPTION
**Pull request by dapengzhang0**
Let child balancer (currently is Round Robin) handle the address list with only healthy addresses for its locality. If no healthy address available, let the child balancer `handleNameResolutionError`.

@voidzcy @zhangkun83 How do you think the approach?

### CODE REVIEW

**Comment by dapengzhang0** [comment]
I discuss with @ejona86 offline who tended to go to TRANSIENT_FAILURE/UNAVAILABLE in the no-healthy-addresses case and in java implementation let the child policy handle name resolution error if the child policy does not support empty addresses, and not to go to IDLE/BUFFERING because that's the case when we are actively doing something to connect not just passively waiting.

**Comment by markdroth** [comment]
Sorry for not replying earlier.  I think the right thing to do is to treat the case where all endpoints are unhealthy exactly the same as if the locality contained no endpoints.  In that case, we should probably be creating a child policy with no addresses, in which case the child policy should report state TRANSIENT_FAILURE.

**Comment by voidzcy** [comment]
Why do we still do this given that we are migrating to `EdsLoadBalancer`? You should just add this implementation there.

**Comment by dapengzhang0** [comment]
@voidzcy `EdsLoadBalancer` includes a refactoring of `LocalityStoreImpl`, not a re-implementation of `LocalityStoreImpl`. Handling health status is an independent PR and should not be blocked by the refactoring PR.

**Comment by voidzcy** [comment]
Whatever, but you would need to do it again in `EdsLoadBalancer`.

**Comment by dapengzhang0** [comment]
@voidzcy To me, the copying logic of `LocalityStoreImpl` inside `EdsLoadBalancer` was unnecessary and a large amount extra effort that does not have much benefit, I could just reuse the existing `LocalityStoreImpl`, as @markdroth said in the meeting.

File: xds/src/main/java/io/grpc/xds/LocalityStore.java [hunk 1]
```diff
@@ -586,33 +573,48 @@ public void run() {
...
+    }
 
-        LocalityLbEndpoints localityInfo = localityInfoMap.get(locality);
-        final List<EquivalentAddressGroup> eags = new ArrayList<>();
-        for (LbEndpoint endpoint: localityInfo.getEndpoints()) {
+    private static void handleEagsOnChildBalancer(
+        Helper childHelper, final LocalityLbInfo localityLbInfo,
+        final LocalityLbEndpoints localityLbEndpoints, final Locality locality) {
+      final List<EquivalentAddressGroup> eags = new ArrayList<>();
+      for (LbEndpoint endpoint: localityLbEndpoints.getEndpoints()) {
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/LocalityStore.java]: nit: space before ` : `.

**Comment by dapengzhang0** [comment]
@voidzcy Yes.

**Pull request merged**